modules/azure/dns/dns.tf

```diff
@@ -23,7 +23,7 @@ variable "zone_name" {
 }
 
 resource "azurerm_dns_zone" "dns_zone" {
-  count               = "${var.root_dns_zone_name == "" ? 0 : 1}"
+  count               = "1"
```

Comment:
This resource is the one that creates a DNS zone in the installation's resource group and needs to be created always.
Using the `root_dns_zone_name` variable to enable or disable the creation of this delegated zone is wrong and leads to an error during the terraform execution.
The variable should be used to just control the creation of the NS records in the delegating zone (such as `azure.gigantic.io`).